agent/structs/structs.go

```diff
@@ -210,6 +213,7 @@ type RPCInfo interface {
 	TokenSecret() string
 	SetTokenSecret(string)
 	HasTimedOut(since time.Time, rpcHoldTimeout, maxQueryTime, defaultQueryTime time.Duration) bool
+	GetMetadata() map[string]string
```

Comment:
This is our trace context carrier for RPC's. We're going to use the TextMapCarrier to push the trace context into every RPC payload